debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/AbstractMysqlFieldReader.java

```diff
@@ -0,0 +1,61 @@
+package io.debezium.connector.mysql.legacy;
+
+import io.debezium.relational.Column;
+import io.debezium.relational.Table;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Types;
+
+/**
+ * Abstract class for decode MySQL return value according to different protocols.
+ *
+ * @author yangjie
+ */
+public abstract class AbstractMysqlFieldReader implements MysqlFieldReader {
```

Comment:
Hum, hum, that's interesting. Did you consciously decide to adjust the legacy implementation of the MySQL connector? The better place for all these changes would be `MySqlSnapshotChangeEventSource` and related code paths. I'm not opposing to _also_ adjusting the legacy implementation (though it is not going to be the focus of engineering going forward), but it should not be _only_ done there.

Reply:
I will fix this on current implementation.